packages/bridge/src/nomic/index.ts

```diff
@@ -36,12 +52,224 @@ export class NomicBridgeProvider implements BridgeProvider {
+        message: "Nomic Bitcoin asset not found in asset list.",
+      });
+    }
+
+    const transferMethod = nomicBtc.transferMethods.find(
+      (method): method is IbcTransferMethod => method.type === "ibc"
+    );
+
+    if (!transferMethod) {
+      throw new BridgeQuoteError({
+        bridgeId: "Nomic",
+        errorType: "UnsupportedQuoteError",
+        message: "IBC transfer method not found for Nomic Bitcoin asset.",
+      });
+    }
+
+    const nomicChain = this.ctx.chainList.find(
+      ({ chain_name }) => chain_name === transferMethod.counterparty.chainName
+    );
+
+    if (!nomicChain) {
+      throw new BridgeQuoteError({
+        bridgeId: "Nomic",
+        errorType: "UnsupportedQuoteError",
+        message: "Nomic chain not found in chain list.",
+      });
+    }
+
+    let swapMessages: Awaited<ReturnType<typeof getSwapMessages>>;
+    let swapRoute:
+      | Awaited<ReturnType<typeof getRouteTokenOutGivenIn>>
+      | undefined;
+
+    if (fromAsset.address !== this.nBTCMinimalDenom) {
+      swapRoute = await getRouteTokenOutGivenIn({
+        assetLists: this.ctx.assetLists,
+        tokenInAmount: fromAmount,
+        tokenInDenom: fromAsset.address,
+        tokenOutDenom: this.nBTCMinimalDenom,
+      });
+
+      swapMessages = await getSwapMessages({
+        coinAmount: fromAmount,
+        maxSlippage: "0.005",
+        quote: swapRoute,
+        tokenInCoinDecimals: fromAsset.decimals,
+        tokenInCoinMinimalDenom: fromAsset.address,
+        tokenOutCoinDecimals: nomicBtc.decimals,
+        tokenOutCoinMinimalDenom: nomicBtc.coinMinimalDenom,
+        userOsmoAddress: fromAddress,
+        quoteType: "out-given-in",
+      });
+    } else {
+      swapMessages = [];
+      swapRoute = undefined;
+    }
+
+    if (!swapMessages) {
+      throw new BridgeQuoteError({
+        bridgeId: "Nomic",
+        errorType: "CreateCosmosTxError",
+        message: "Failed to get swap messages.",
+      });
+    }
+    const nomicBridgeAsset: BridgeAsset = {
+      address: nomicBtc.coinMinimalDenom,
+      decimals: nomicBtc.decimals,
+      denom: nomicBtc.symbol,
+      coinGeckoId: nomicBtc.coingeckoId,
+    };
+    const ibcProvider = new IbcBridgeProvider(this.ctx);
+
+    const transactionDataParams: GetBridgeQuoteParams = {
+      ...params,
+      fromAmount: !!swapRoute
+        ? swapRoute.amount.toCoin().amount
+        : params.fromAmount,
+      fromAsset: nomicBridgeAsset,
+      toChain: {
+        chainId: nomicChain.chain_id,
+        chainType: "cosmos",
+        chainName: nomicChain.pretty_name,
+      },
+      toAsset: nomicBridgeAsset,
+      toAddress: deriveCosmosAddress({
+        address: fromAddress,
+        desiredBech32Prefix: "nomic",
+      }),
+    };
+
+    const [ibcTxMessages, estimatedTime] = await Promise.all([
+      ibcProvider.getTxMessages({
+        ...transactionDataParams,
+        memo: destMemo,
+      }),
+      ibcProvider.estimateTransferTime(
```

Comment:
Does the estimated time include confirmation on Bitcoin network side? Maybe we need to hardcode something for that. IDK how nomic withdrawals work though

Reply:
Yeah, I’m coordinating with Nomic to figure out how we can track the progress and status of withdrawals. For now, I think it’s best to set the estimated time based on IBC, as that’s the only timeframe we are certain about.